scripts/rpc-maas-tool.py

```diff
@@ -13,199 +13,600 @@
+                            action="append",
+                            help='A check that should not be present'
+                                 'can be specified multiple times',
+                            default=[])
+        self.args = parser.parse_args()
+
+    def main(self):
+        if self.rpcm.conn is None:
+            LOGGER.error("Unable to get a client to MaaS, exiting")
+            sys.exit(1)
+
+        dd = {'list-alarms': self.alarms,
+              'run-checks': self.run_checks,
+              'list-checks': self.checks,
+              'compare-checks': self.compare_checks,
+              'checks-without-alarms': self.checks_without_alarms,
+              'compare-alarms': self.compare_alarms,
+              'overview': self.overview,
+              'verify': self.verify,
+              'delete': self.delete,
+              'remove-defunct-checks': self.remove_defunct_checks,
+              'remove-defunct-alarms': self.remove_defunct_alarms}
+        result = dd[self.args.command]()
+        if result is None:
+            return 0
         else:
-            conn = driver(user, api_key)
-    if not conn and config.has_section('api'):
-        try:
-            url = config.get('api', 'url')
-            token = config.get('api', 'token')
-        except Exception as e:
-            print(e)
+            return result
+
+    def checks_without_alarms(self):
+        """list checks with no alarms"""
+        no_alarms = [c for c in self.rpcm.get_checks(self.args.checkprefix)
+                     if not c.alarms and c.type != 'remote.ping']
+        if no_alarms:
+            LOGGER.info("The following checks have 0 alarms "
+                        "registered with the maas api:")
+            self._write(no_alarms)
+            return 1
+
+    def excluded_checks(self):
+        """List checks that are present but shouldn't be"""
+        present_but_excluded_checks = []
+        checks = self.rpcm.get_checks(self.args.entityprefix)
+        for check in checks:
+            for exclude in self.args.excludedcheck:
+                # ping is a default check and may not have an alarm
+                if exclude in check.label:
+                    present_but_excluded_checks.append(check)
+        if present_but_excluded_checks:
+            LOGGER.info("The following checks are in the excluded_checks list"
+                        " but are still present in the API:")
+            self._write(present_but_excluded_checks)
+            return 1
+
+    def compare_checks(self):
+        """Compare checks
+
+        Check that all checks found in config files are registered with The
+        maas api.
+        """
+
+        api_checks = self.rpcm.get_checks(self.args.checkprefix)
+        missing_checks = [configcheck for configcheck in
+                          self.rpcmac.checks.keys()
+                          if configcheck not in
+                          [apicheck.label for apicheck in api_checks]]
```

Comment:
sets are nice for this type of operation, e.g.

set(self.rpcmac.checks) - set(apicheck.label for apicheck in api_checks)